packages/new-deploy/scripts/deploy.ts

```diff
@@ -28,24 +28,24 @@ async function run() {
     deployContext.registerSigner(network, signer);
     deployContext.overrides.set(network, OVERRIDES[network]);
   }
-  // run the deploy script
   const outputDir = './output';
   try {
-    const governanceBatch = await deployContext.deployAndRelinquish();
+    // run the deploy script
+    await deployContext.deployAndRelinquish();
 
+    // output the updated config & verification
     outputConfigAndVerification(outputDir, deployContext);
 
-    if (governanceBatch && !governanceBatch.isEmpty()) {
-      // build & write governance batch
-      await governanceBatch.build();
-      fs.writeFileSync(
-          `${outputDir}/governanceTransactions.json`,
-          JSON.stringify(governanceBatch, null, 2),
-      );
-    }
+    // output the governance transactions when the deploy succeeds
+    // (if the deploy script throws, the callBatch will be
+    // re-generated idempotently when the script ultimately succeeds)
+    await outputCallBatch(outputDir, deployContext);
```

Comment:
Could we use methods and standalone functions, because it feels like a step backwards towards the old deploy

Reply:
hmmm yeah, all of our code for reading / writing from local files is right now in a functional way rather than the Object-Oriented, but we could refactor to move it into the object perhaps